pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -149,6 +149,13 @@ let
       NF_TABLES_ARP               = whenAtLeast "4.17" yes;
       NF_TABLES_IPV6              = whenAtLeast "4.17" yes;
       NF_TABLES_BRIDGE            = whenBetween "4.17" "5.3" yes;
+
+      # needed for ss
+      INET_DIAG         = yes;
+      INET_TCP_DIAG     = module;
+      INET_UDP_DIAG     = module;
+      INET_RAW_DIAG     = whenAtLeast "4.14" module;
+      INET_DIAG_DESTROY = whenAtLeast "4.9" yes;
```

Comment:
Do these DIAG options incur a performance penalty we need to be aware of?

Reply:
I don't think there is a performance penalty. I looked at

https://github.com/torvalds/linux/blob/master/net/ipv4/udp_diag.c
https://github.com/torvalds/linux/blob/master/net/ipv4/tcp_diag.c
https://github.com/torvalds/linux/blob/master/net/ipv4/raw_diag.c

and it looks like they just provide additional functions. The init handler doesn't appear to change any network stack behavior.

`INET_DIAG_DESTROY` should be fine too; it is not referenced outside those `_diag` modules.

Reply:
@grahamc it only extends the netlink API of the kernel. Looks fine to me.